client/blocks/importer/wordpress/upgrade-plan/upgrade-plan-details.tsx

```diff
@@ -1,24 +1,220 @@
+					onClick={ onMonthlyPlanClick }
+				>
+					{ __( 'Pay monthly' ) }
+				</Button>
+				<Button
+					borderless
+					className={ clsx( { selected: selectedPlan === PLAN_BUSINESS } ) }
+					onClick={ onAnnualPlanClick }
+				>
+					{ __( 'Pay annually' ) }
+				</Button>
+			</ButtonGroup>
+		</div>
+	);
+};
+
+const Badge = styled.div< { isForIntroOffer?: boolean; isHidden?: boolean } >`
+	text-align: center;
+	white-space: nowrap;
+	font-size: 0.75rem;
+	line-height: 1.25rem;
+	border-radius: 4px;
+	height: 21px;
+	display: inline-block;
+	width: fit-content;
+	letter-spacing: 0.2px;
+	font-weight: 500;
+	text-align: center;
+	padding: 0 12px;
+	background-color: #e6d1ff;
+	color: #350e63;
+	text-transform: none;
+`;
+
+const PricesGroup = styled.div`
+	justify-content: flex-end;
+	display: flex;
+	flex-direction: row-reverse;
+	align-items: flex-end;
+	gap: 4px;
+`;
+
+interface PriceWithIntroductoryOfferProps {
+	originalMonthlyPrice: number;
+	introOfferMonthlyPrice: number;
+	currencyCode?: string;
+}
+
+const PriceWithIntroductoryOffer = ( props: PriceWithIntroductoryOfferProps ) => {
+	const { originalMonthlyPrice, introOfferMonthlyPrice, currencyCode } = props;
+	const translate = useTranslate();
+
+	return (
+		<>
+			<Badge>{ translate( 'One time offer' ) }</Badge>
+			<PricesGroup>
+				<PlanPrice
+					rawPrice={ originalMonthlyPrice }
+					currencyCode={ currencyCode }
+					original
+					isSmallestUnit
+				/>
+				<PlanPrice rawPrice={ introOfferMonthlyPrice } currencyCode={ currencyCode } />
+			</PricesGroup>
+		</>
+	);
+};
+
+interface PlanPriceOfferProps {
+	plan?: Plan;
+	currencyCode?: string;
+	originalMonthlyPrice?: number;
+	introOfferMonthlyPrice?: number;
+	formattedOriginalPrice?: string;
+	formattedIntroOfferPrice?: string;
+	introOfferEnabled: boolean;
+	isIntroOfferComplete: boolean;
+}
+
+const PlanPriceOffer = ( props: PlanPriceOfferProps ) => {
+	const translate = useTranslate();
+
+	const {
+		plan,
+		formattedOriginalPrice,
+		formattedIntroOfferPrice,
+		introOfferEnabled,
+		isIntroOfferComplete,
+		introOfferMonthlyPrice,
+		originalMonthlyPrice,
+		currencyCode,
+	} = props;
+
+	const showIntroOffer =
+		introOfferEnabled && ! isIntroOfferComplete && introOfferMonthlyPrice && originalMonthlyPrice;
+
+	const price = showIntroOffer ? (
+		<PriceWithIntroductoryOffer
+			originalMonthlyPrice={ originalMonthlyPrice }
+			introOfferMonthlyPrice={ introOfferMonthlyPrice }
+			currencyCode={ currencyCode }
+		/>
+	) : (
+		<PlanPrice rawPrice={ originalMonthlyPrice } currencyCode={ currencyCode } isSmallestUnit />
+	);
+
+	if ( plan && showIntroOffer && formattedIntroOfferPrice ) {
+		plan.getBillingTimeFrame = () =>
+			translate(
+				'per month, %(discounted)s billed annually for the first year, %(original)s atferwards',
+				{
+					args: {
+						discounted: formattedIntroOfferPrice,
+						original: formattedOriginalPrice || '',
+					},
+				}
+			);
+	}
```

Comment:
Overwriting `plan.getBillingTimeFrame()` feels like it could easily have major unintended side-effects! I would strongly recommend we push the value into a local `const` instead.

Also, the text has a typo `atferwards` => `afterwards`, and it would be well worth adding translator comments and/or changing the variable names to describe what the values are. (The exact comment format may depend on whether you're using `__` or `translate()`, but I'll use the `translate()` format here.)
```suggestion
	const billingTimeFrame = plan && showIntroOffer && formattedIntroOfferPrice
		? translate(
			'per month, %(discountedPrice)s billed annually for the first year, %(originalPrice)s afterwards',
			{
				args: {
					discountedPrice: formattedIntroOfferPrice,
					originalPrice: formattedOriginalPrice || '',
				},
				comment: 'discountedPrice is a formatted price like $150; originalPrice is a formatted price like $200',
			}
		) : plan?.getBillingTimeFrame();		
```